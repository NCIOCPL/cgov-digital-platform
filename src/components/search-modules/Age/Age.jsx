import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Fieldset, TextInput } from '../../atomic';

const Age = ({ handleUpdate }) => {
  const age = useSelector(store => store.form.a);
  const [errorMsg, setErrorMsg] = useState('');

  const validateAgeEntry = () => {
    let parsedAge = parseInt(age);
    if (
      Number.isNaN(parsedAge) ||
      parsedAge > 120 ||
      parsedAge < 0
    ) {
      setErrorMsg('Please enter a number between 1 and 120.');
    }else {
      setErrorMsg('');
    }
  };

  return (
    <Fieldset
      id="age"
      legend="Age"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
    >
      <TextInput
        action={e => handleUpdate(e.target.id, e.target.value)}
        id="a"
        value={age}
        label="age"
        labelHidden
        errorMessage={errorMsg}
        inputHelpText="Your age helps determine which trials are right for you."
        maxLength={3}
        onBlur={validateAgeEntry}
      />
    </Fieldset>
  );
};

export default Age;
