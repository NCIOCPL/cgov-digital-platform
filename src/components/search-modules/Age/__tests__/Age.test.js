import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { render } from 'enzyme';
import Age from '../Age';

const mockStore = configureMockStore();

const defaultProps = {};

const initialState = {
  form: {
    a: '',
  },
};

const setup = (enzymeMethod = render, props = {}, initialState = {}) => {
  const store = mockStore(initialState);
  const component = enzymeMethod(
    <Provider store={store}>
      <Age {...props} />
    </Provider>
  );
  return component;
};

describe('Age', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = setup(render, defaultProps, initialState);
      expect(component).toMatchSnapshot();
    });
  });
});
