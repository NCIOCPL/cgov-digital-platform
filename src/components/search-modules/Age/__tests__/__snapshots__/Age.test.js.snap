// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--age"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Age
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
    <div
      class=""
    >
      <label
        class="cts-label"
        for="a"
        id="a-label"
      >
        Enter the age of the participant.
      </label>
      <input
        aria-labelledby="a-label"
        aria-required="false"
        class="cts-input  "
        id="a"
        maxlength="3"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
